Dockerizing App for Google Cloud Run

jbh14 opened the pull request:
- Dockerfile to build image for deployment to Google Cloud Run
- Small update to HomePage.js to allow reading in a "fetch URL" from local .env (this is gitignored) if wanted to test with locally running backend, or otherwise default to our deployed backend URL.  If preferred, could also attempt to read backend URL from .env as well.

dscottd7 commented:
@jbh14 I think this line should be the other way around, where the deployed frontend link is hidden in the `.env` file and `localhost` is the fallback address:

```
// reads URL from .env - be sure to list 'REACT_APP_BACKEND_URL=http://localhost:8000' there if wanting to test fetching from locally running backend app
  const FETCH_URL = process.env.REACT_APP_BACKEND_URL || 'https://quibble-backend-64265842032.us-central1.run.app';
```

In the .env file, we should have:

`REACT_APP_BACKEND_URL=https://quibble-backend-64265842032.us-central1.run.app`

And then in the frontend code, we should have:

`const FETCH_URL = process.env.REACT_APP_BACKEND_URL || 'http://localhost:8000';`

I don't think we want to have the URL for the frontend or the backend in either of the codebases.

jbh14 commented:
@dscottd7 good suggestion - I've implemented (and updated a few ignore files to ignore env), and this should be fully set-able when deploying to Google Cloud Run as well.  I may wait to do that until we fix the loading bar issue in "main" without hitting an error, but we should be OK to merge this change while working on a fix for that.

dscottd7 reviewed (approved):
I fixed the merge conflicts and am approving.
